app/javascript/mixins/mixin-select-shared.js

```diff
@@ -0,0 +1,205 @@
+  ],
+
+  props: {
+    placeholder: {
+      default: '',
+      type: String
+    }
+  },
+
+  data () {
+    return {
+      isActive: false,
+      highlightedOptionIndex: -1,
+      searchTerm: '',
+      dropdownId: 'v-select-dropdown-' + this.config.id,
+      dropdownOptionsName: 'v-select-dropdown-input' + this.config.id,
+      toggleId: 'v-select-toggle-' + this.config.id,
+      searchId: 'v-select-search-' + this.config.id,
+      searchResetId: 'v-select-search-reset-' + this.config.id
+    }
+  },
+
+  computed: {
+    hasKeyboardFocus () {
+      return this.highlightedOptionIndex >= 0
+    },
+
+    highlightedOptionId () {
+      if (this.isActive && this.filteredOptions.length && this.hasKeyboardFocus) {
+        return this.getOptionInputId(this.filteredOptions[this.highlightedOptionIndex])
+      }
+
+      return null
+    },
+
+    isDisabled () {
+      return !this.options.length
+    },
+
+    getPlaceholderText () {
+      return this.placeholder || ''
+    },
+
+    showOptions () {
+      return this.isActive && Boolean(this.filteredOptions.length)
+    },
+
+    showResetIcon () {
+      return this.searchTerm && this.isActive
+    },
+
+    showSearchIcon () {
+      return this.hasSearchIcon && !this.showResetIcon
+    }
+  },
+
+  watch: {
+    searchTerm () {
+      this.resetHighlightedIndex()
+    },
+
+    highlightedOptionId (id) {
+      if (id !== null) {
+        this.scrollToHighlightedOption()
+      }
+    }
+  },
+
+  created () {
+    this.initializeSelectedInternal()
+  },
+
+  mounted () {
+    this.addTabFromSearchListener()
+    this.addArrowKeyListeners()
+    this.addTabForwardFromResetListener()
+  },
+
+  methods: {
+    toggleSelect (e) {
+      if (this.options.length && !this.isActive) {
+        this.openSelect(e)
+      } else {
+        this.closeSelect(e)
+      }
+    },
+
+    isHighlighted (index) {
+      return index === this.highlightedOptionIndex
+    },
+
+    resetHighlightedIndex() {
+      this.highlightedOptionIndex = -1
+    },
+
+    getOptionFromId (id) {
+      const filteredOptions = this.options.filter( option => {
+        return option.id === id
+      })
+
+      return filteredOptions.length ? filteredOptions[0] : null
+    },
+
+    getOptionInputId (option) {
+      const friendlyId = option.id.toString().replace(/[ ()]/g, '-')
+
+      return `option-${this.config.id}-${friendlyId}`
+    },
+
+    matchesSearchTerm (option) {
+      const regex = new RegExp(`${this.searchTerm}`, 'i')
+      const match = option.name.toString().match(regex)
+
+      return !this.searchTerm || match
+    },
+
+    emitSelection () {
+      this.$emit('update:selected-option', this.selectedInternal)
+    },
+
+    resetSearchTermClick () {
+      this.resetSearchTerm()
+      this.focusSearch()
+    },
+
+    resetSearchTerm () {
+      this.searchTerm = ''
+    },
+
+    scrollToHighlightedOption () {
+      this.$el.querySelector('#' + this.highlightedOptionId).scrollIntoView()
```

Comment:
When trying to use up and down arrow keys to navigate the dropdown items I get the following error related to this line: ```Error in callback for watcher "highlightedOptionId": "TypeError: this.$el.querySelector(...) is null"```

Not sure if related, but dropdown items are not highlighted when navigating using the arrow keys; however, I can see it's actually selecting those anyway after hitting enter.

Reply:
@WKocur has resolved this issue